jol-core/src/main/java/org/openjdk/jol/info/FieldLayout.java

```diff
@@ -93,13 +93,14 @@ public String name() {
     }
 
     public String shortFieldName() {
+        String classShortName = classShortName();
+        return classShortName +  "." + name();
+    }
+
+    public String classShortName() {
         String cl = hostClass();
         int idx = cl.lastIndexOf(".");
-        if (idx != -1 && idx < cl.length()) {
-            return cl.substring(idx + 1) + "." + f.name();
-        } else {
-            return cl + "." + name();
-        }
+        return idx != -1 && idx < cl.length() ? cl.substring(idx + 1) : cl;
```

Comment:
Took me a while to recognize the ternary statement here. Should at least have parentheses? 

```
 return (idx != -1 && idx < cl.length()) ? cl.substring(idx + 1) : cl;
```

Or maybe don't do this at all, and leave the if block:

```
if (idx != -1 && idx < cl.length()) {
  return cl.substring(idx + 1)
} else {
  return cl;
}
```